src/industrytype.h

```diff
@@ -106,15 +106,15 @@ struct IndustrySpec {
 	IndustryType conflicting[3];                ///< Industries this industry cannot be close to
 	uint8_t check_proc;                            ///< Index to a procedure to check for conflicting circumstances
 	std::array<CargoID, INDUSTRY_NUM_OUTPUTS> produced_cargo;
-	std::variant<CargoLabel, MixedCargoType> produced_cargo_label[INDUSTRY_NUM_OUTPUTS];
+	std::variant<CargoLabel, MixedCargoType> produced_cargo_label[INDUSTRY_ORIGINAL_NUM_OUTPUTS];
 	uint8_t production_rate[INDUSTRY_NUM_OUTPUTS];
 	/**
 	 * minimum amount of cargo transported to the stations.
 	 * If the waiting cargo is less than this number, no cargo is moved to it.
 	 */
 	uint8_t minimal_cargo;
 	std::array<CargoID, INDUSTRY_NUM_INPUTS> accepts_cargo; ///< 16 accepted cargoes.
```

Comment:
Should this line also be changed? (Code and comment)

Reply:
All industries can have 16 input/output.
The change in this PR is specifically for cargo_label because it's only used for vanilla industries as a compatibility layer.
If you look for `produced_cargo_label` usage, you'll see it's invalidated as soon as the industry production is modified by newgrf, same for `accepts_cargo_label`.